search-sentiments-map/test/trends-test.js

```diff
@@ -0,0 +1,228 @@
+
+    beforeEach(() => {
+      datastoreEntities = [];
+
+      let mockDailyTrendsData = JSON.stringify({
+        'default': {
+          'trendingSearchesDays': [
+            {
+              'trendingSearches': 'mockTrendingSearches',
+            }
+          ],
+        },
+      });
+
+      mockTrendsByCountry = [];
+      for (let i = 0; i < countryJson.length; i++) {
+        mockTrendsByCountry.push({
+          country: countryJson[i].id,
+          trends: 'mockTrends',
+        });
+      }
+
+      // Stub the API call to googleTrends.
+      sinon.stub(googleTrends, 'dailyTrends').resolves(mockDailyTrendsData);
+
+      // Stub the `constructCountryTrendsJson` function because it is tested 
+      // separately.
+      sinon.stub(trends, 'constructCountryTrendsJson')
+          .callsFake((trendingSearches, countryCode) => {
+        return {
+          country: countryCode,
+          trends: 'mockTrends',
+        }
+      });
+
+      // Stub the `deleteAncientTrend` function because it is tested separately.
+      deleteAncientTrendsStub = 
+          sinon.stub(trends, 'deleteAncientTrend')
+              .resolves('Not interested in the output');
+
+      // Stub the `getGlobalTrends` function because it is tested separately.
+      getGlobalTrendsStub = 
+          sinon.stub(trends, 'getGlobalTrends').callsFake(() => {
+            return 'mockGlobalTrends';
+          });
+
+      // Stub calls to the datastore.
+      sinon.stub(Datastore.prototype, 'key').callsFake(() => {
+        return 'fakeKey';
+      });
+
+      // Stub calls to the datastore.
+      sinon.stub(Datastore.prototype, 'save').callsFake((entity) => {
+        datastoreEntities.push(entity);
+      });
+    });
+
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('should get daily trends and save in datastore', async function() {
+      await trends.updateDailyTrends();
+      assert.equal(deleteAncientTrendsStub.callCount, 1);
+      assert.equal(getGlobalTrendsStub.callCount, 1);
+      assert.equal(datastoreEntities[0].key, 'fakeKey');
+      assert.equal(
+          datastoreEntities[0].data.globalTrends, 'mockGlobalTrends');
+      assert.deepEqual(
+          datastoreEntities[0].data.trendsByCountry, mockTrendsByCountry);
+    });
+  });
+
+  describe('ConstructCountryTrendsJson', function() {
+    it ('should return correctly formatted country trends', function() {
+      let countryCode = 'US';
+      let trendingSearches = [
+        {
+          title: {
+            query: 'Example Topic 1',
+            exploreLink: '/trends/exploreLink1',
+          },
+          formattedTraffic: '200K+',
+          articles: [
+            {
+              title: 'Example Topic 1 Article 1 Title',
+              url: 'https://example_topic_1_article_1',
+            }, {
+              title: 'Example Topic 1 Article 2 Title',
+              url: 'https://example_topic_1_article_2',
+            },
```

Comment:
I don't think this comma should be here because this is in an array, not an object.

Reply:
ok